Services/Administration/OJS.Services.Administration.Business/Implementations/ProblemGroupsBusinessService.cs

```diff
@@ -76,6 +79,33 @@ await sourceContestProblemGroups
             return ServiceResult.Success;
         }
 
+        public async Task ReevaluateProblemsAndProblemGroupsByOrderBy(int contestId)
+        {
+            var orderByIndex = 0;
+
+            this.problemGroupsData.GetAllNonDeletedByContest(contestId)
+                .OrderBy(pg => pg.OrderBy)
+                .ForEach(pg =>
+                {
+                    pg.OrderBy = ++orderByIndex;
+                    this.problemGroupsData.Update(pg);
+                });
+
+            await this.problemGroupsData.SaveChanges();
+
+            orderByIndex = 0;
+
+            this.contestsData.GetProblemsById(contestId)
+                                .OrderBy(p => p.OrderBy)
+                                .ForEach(p =>
+                                {
+                                    p.OrderBy = ++orderByIndex;
+                                    this.problemsData.Update(p);
+                                });
+
+            await this.problemsData.SaveChanges();
```

Comment:
This could be extracted into a generic service. Pass IEnumerable<TEntity> where TEntity : IOrderable to a method `ReevaluateOrder( IEnumerable<TEntity> entities)` and inject IDataService<TEntity> in the service.